integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java

```diff
@@ -250,20 +251,30 @@ public void requestContainers() throws Exception {
     // is initialized to the number of requests made. (2) is then achieved by counting down whenever
     // a container is allocated, and waiting here for the number of outstanding requests to hit 0.
     int round = 0;
-    while (mWorkerHosts.size() < mNumWorkers && round < MAX_WORKER_CONTAINER_REQUEST_ROUNDS) {
-      requestWorkerContainers();
-      LOG.info("Waiting for {} worker containers to be allocated",
-          mOutstandingWorkerContainerRequestsLatch.getCount());
-      // TODO(andrew): Handle the case where something goes wrong and some worker containers never
-      // get allocated. See TACHYON-1410
-      mOutstandingWorkerContainerRequestsLatch.await();
-      round ++;
+    while (mWorkerHosts.size() < mNumWorkers && round ++ <
+        Constans.INTEGRATION_YARN_WORKER_CONTAINER_REQUEST_ROUNDS_MAX) {
```

Comment:
`Constants.INTEGRATION_YARN_WORKER_CONTAINER_REQUEST_ROUNDS_MAX` is a string rather than a integer. you need to get this value under this name from conf